volume-providers/vpc/provider/get_volume_attachment.go

```diff
@@ -68,10 +78,15 @@ func (vpcs *VPCSession) getVolumeAttachmentByVolumeID(volumeAttachmentRequest mo
 	vpcs.Logger.Info("Getting VolumeAttachmentList from VPC provider...")
 	var volumeAttachmentList *models.VolumeAttachmentList
 	var err error
-	err = retry(vpcs.Logger, func() error {
+	err = vpcs.APIRetry.FlexyRetry(vpcs.Logger, func() (error, bool) {
 		volumeAttachmentList, err = vpcs.APIClientVolAttachMgr.ListVolumeAttachment(&volumeAttachmentRequest, vpcs.Logger)
-		return err
+		// Keep retry, until we get the proper volumeAttachmentRequest object
+		if err != nil {
+			return err, skipRetryForAttach(err, vpcs.Config.IsIKS)
```

Comment:
1. skipRetryForAttach(err, vpcs.Config.IsIKS)  , can't this be generic SKIP code, why we say it "attach" specific ?
cant error code to be skipped , passed as part of the Retry parameter. It makes the code complicated.
2. Also does this applicable for "detach" ?  i dont see that,    is direct detach/IKS detach return same Error code
3. These IKS specific attach is part of provider/util , it could have been part of IKS provider/VPC provider
4. Overall i see the IKS and attach checks have made this Retry complicated.  and also specific error code skip  logic. May be these codes need a future optimizations.
5. The constant retry and exponential retry could have put under the same method and based upon the parameters, the logic could have been tried.

Reply:
1- as of now its only used for volume attach related thats why once we move for all then we can change its name
2- As of now detach does not required this, only attach require this, error are different but this PR is mainly for retry when to retry and when not to retry thats why this method is not useful in case of detach. 
3- Actually there is only one provider which handles both IKS and VPC so its not completely two providers, I think it looks good only improvements I see is to have same type of errors returns from these two IKS and RIaaS in the lib itself.
4- I did not see complexity in this may be you don't understand completely or once we move adopt retry for all then it will be very simple.
5- may be, I discussed with Akash and we finalise to have separate method.

overall may be some improvements needed in terms of error handling, we can take it in another phase or PR. Let me know if anything breaking by this PR??